spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/support/DataFlowSwaggerApiDocsJsonDecodeFilter.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2016-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.server.support;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.text.StringEscapeUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+import org.springframework.web.util.ContentCachingResponseWrapper;
+
```

Comment:
This filter is required because there is an issue similar to this one: https://github.com/springdoc/springdoc-openapi/issues/624. The api-docs endpoints returns escaped JSON. Because Spring Cloud Data Flow relies on escaped JSON content for example to show the output logs of a task execution I implemented this filter.

Reply:
Thanks for the details @klopfdreh . We can basically copy/paste your comment as the javadoc - well stated. 

> The endpoints springdoc.swagger-ui.configUrl and springdoc.api-docs.path return escaped JSON.

Is this the default behavior for Springdoc or is this because of some config settings in SCDF to satisfy the "Spring Cloud Data Flow relies on escaped JSON content" requirement?

Reply:
Normally the endpoint of api-docs.path returns a valid JSON which is the default behavior. I assume that because SCDF communicates through JSON with the backend and requires it to be escaped to parse it in Angular this behavior is requires for SCDF. This is the issue here and the reason why the filter is required